core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -169,7 +160,7 @@ private[spark] object ClosureCleaner extends Logging {
 
   private def instantiateClass(cls: Class[_], outer: AnyRef, inInterpreter: Boolean): AnyRef = {
     // logInfo("Creating a " + cls + " with outer = " + outer)
-    if (!inInterpreter) {
+    if (inInterpreter) {
```

Comment:
@preeze I believe inverting this is incorrect. We want to instantiate the closures directly instead of through reflection normally, so we shouldn't flip this. In the worst case not changing this will have the same behavior as before.